src/Http/Middleware/CheckClientCredentialsForAnyScope.php

```diff
@@ -2,104 +2,45 @@
 {
     /**
-     * The Resource Server instance.
+     * Validate token credentials.
      *
-     * @var \League\OAuth2\Server\ResourceServer
-     */
-    protected $server;
-
-    /**
-     * Token Repository.
-     *
-     * @var \Laravel\Passport\TokenRepository
-     */
-    protected $repository;
-
-    /**
-     * Create a new middleware instance.
-     *
-     * @param  \League\OAuth2\Server\ResourceServer  $server
-     * @param  \Laravel\Passport\TokenRepository  $repository
+     * @param  \Laravel\Passport\Token  $token
      * @return void
+     * @throws \Illuminate\Auth\AuthenticationException
      */
-    public function __construct(ResourceServer $server, TokenRepository $repository)
-    {
-        $this->server = $server;
-        $this->repository = $repository;
-    }
-
-    /**
-     * Handle an incoming request.
-     *
-     * @param  \Illuminate\Http\Request  $request
-     * @param  \Closure  $next
-     * @param  mixed  ...$scopes
-     * @return mixed
-     * @throws \Illuminate\Auth\AuthenticationException|\Laravel\Passport\Exceptions\MissingScopeException
-     */
-    public function handle($request, Closure $next, ...$scopes)
+    protected function validateCredentials($token)
     {
-        $psr = (new PsrHttpFactory(
-            new ServerRequestFactory,
-            new StreamFactory,
-            new UploadedFileFactory,
-            new ResponseFactory
-        ))->createRequest($request);
-
-        try {
-            $psr = $this->server->validateAuthenticatedRequest($psr);
-        } catch (OAuthServerException $e) {
+        if (! $token || $token->client->firstParty()) {
             throw new AuthenticationException;
         }
-
-        if ($this->validate($psr, $scopes)) {
-            return $next($request);
-        }
-
-        throw new MissingScopeException($scopes);
     }
 
     /**
-     * Validate the scopes and token on the incoming request.
+     * Validate token credentials.
      *
-     * @param  \Psr\Http\Message\ServerRequestInterface $psr
+     * @param  \Laravel\Passport\Token  $token
      * @param  array  $scopes
-     * @return bool
-     * @throws \Illuminate\Auth\AuthenticationException
+     * @return void
+     * @throws \Laravel\Passport\Exceptions\MissingScopeException
      */
-    protected function validate($psr, $scopes)
+    protected function validateScopes($token, $scopes)
     {
-        $token = $this->repository->find($psr->getAttribute('oauth_access_token_id'));
-
-        if (! $token || $token->client->firstParty()) {
-            throw new AuthenticationException;
-        }
-
         if (in_array('*', $token->scopes)) {
-            return true;
+            return;
         }
 
         foreach ($scopes as $scope) {
             if ($token->can($scope)) {
-                return true;
+                return;
             }
         }
 
-        return false;
+        throw new MissingScopeException($scopes);
```

Comment:
Both `CheckClientCredentials` and `CheckClientCredentialsForAnyScope` are the same except

* `CheckClientCredentials` throws an exception of failure and return `void`.
* `CheckClientCredentialsForAnyScope` return `bool` and `handle` method convert `false` to an exception.

This changes remove the different (making both identical).